src/mongocxx/uri.cpp

```diff
@@ -166,6 +167,18 @@ static stdx::optional<bool> _bool_option(mongoc_uri_t* uri, std::string opt_name
     return bson_iter_bool(&iter);
 }
 
+stdx::optional<bsoncxx::document::view> uri::credentials() {
+    const bson_t* options_bson = libmongoc::uri_get_credentials(_impl->uri_t);
+    const uint8_t* data = bson_get_data(options_bson);
+    uint32_t len = options_bson->len;
+
+    if (len) {
+        return bsoncxx::document::view(data, len);
+    } else {
+        return {};
+    }
```

Comment:
```suggestion
    
    return bsoncxx::document::view(data, options_bson->len);
```
Length check should not be necessary.